lib/internal/main/compile.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const {
+  prepareMainThreadExecution,
+  markBootstrapComplete,
+} = require('internal/process/pre_execution');
+const { copyFile, readFile } = require('fs/promises');
+const { emitExperimentalWarning } = require('internal/util');
+const { getOptionValue } = require('internal/options');
+const { inject } = require('internal/deps/postject/postject');
+
+prepareMainThreadExecution(false);
+markBootstrapComplete();
+
+emitExperimentalWarning(`${process.argv0} --compile`);
+
+(async () => {
+  const compilePath = getOptionValue('--compile');
+  const resourceData = await readFile(process.argv[1]);
+  await copyFile(process.execPath, compilePath);
+  await inject(compilePath, 'NODE_JS_CODE', resourceData);
```

Comment:
This creates an unfortunate race condition if `process.execPath` and/or `compilePath` gets deleted/overwritten when the process is running, could we avoid that?

Reply:
ditto